docs/rules.rst

```diff
@@ -2,10 +2,140 @@
 List of rules
 ****************
 
+.. Esp when writing short descriptions it'd be very handy to link to a glossary, instead of saying stuff like ``except BaseException/trio.Cancelled/anyio.get_cancelled_exc_class()/asyncio.exceptions.CancelledError``
+   it also allows easier use of library-specific terminology without forcing people to know all libraries by heart.
+   It should probably have it's own page in the long run
+
+Glossary
+========
+
+.. _timeout_context:
+
+timeout context
+---------------
+Colloquially referred to as a "Timeout" or "CancelScope". A context manager that enforces a timeout on a block of code. Trio/AnyIO timeout functions are implemented with a ``CancelScope``, but you can also directly use ``CancelScope`` as a context manager.
```

Comment:
```suggestion
.. _cancel_scope:

cancel scope
------------
A cancel scope is a context manager which can request the library cancels
whatever task is executing in the body of the ``with`` (or ``async with``)
block.  A cancel scope is the key component of a timeout context, and used in
TaskGroups / Nurseries to cancel any remaining child tasks if one raises an
exception.  Trio and AnyIO have an explicit ``CancelScope`` type; in asyncio
they are implicit.

.. _timeout_context:

timeout context
---------------
A context manager that enforces a timeout on a block of code, by cancelling it
after a specified duration or at a preset time.  The timeout can also be
rescheduled after creation.
```

Reply:
I guess my only worry here is the rule blurb saying "don't do X in cancelscope" but it's triggering on code that doesn't mention cancelscopes. But with the glossary mentioning nurseries/taskgroups/timeouts, and detailed rules pages also doing so, we're probably fine.